Neos.Flow/Classes/Validation/Validator/FileExtensionValidator.php

```diff
@@ -0,0 +1,72 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Flow\Validation\Validator;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\ResourceManagement\ResourceMetaDataInterface;
+use Psr\Http\Message\UploadedFileInterface;
+
+/**
+ * The given $value is has one of the allowed file extensions
+ * Note: a value of NULL or empty string ('') are considered valid
```

Comment:
this comment doesnt seem to be correct ;)
```php
        if ($fileExtension === null || $fileExtension === '') {
             $this->addError('The file has no file extension.', 1677934932);
             return;
         }
```

Reply:
The abstract validator already ensures that empty values are ignored by default.

this line handles cases where there was a resource/upload but the file name was empty or no extension was found.

This validator should be combined with a NotEmptyValidator in that case.

Reply:
Ah okay then i misread that sorry ^^